sr/robot3/raw_serial.py

```diff
@@ -63,9 +65,20 @@ def __init__(
             serial_port,
             baudrate=baudrate,
             timeout=0.5,  # 500ms timeout
+            do_not_open=True,
         )
-
-        sleep(2)  # Wait for boards to reset after connecting
+        try:
```

Comment:
try: except is duplicate code with the `_reconnect` method, could we use this instead